kubernetes/resource_kubernetes_env.go

```diff
@@ -308,13 +325,22 @@ func resourceKubernetesEnvRead(ctx context.Context, d *schema.ResourceData, m in
 	return nil
 }
 
-func getResponseEnvs(u *unstructured.Unstructured, containerName string, kind string) ([]interface{}, error) {
+func getResponseEnvs(u *unstructured.Unstructured, containerName string, kind string, isInitContainer bool) ([]interface{}, error) {
 	var containers []interface{}
 
-	containers, _, _ = unstructured.NestedSlice(u.Object, "spec", "template", "spec", "containers")
+	if isInitContainer {
+		containers, _, _ = unstructured.NestedSlice(u.Object, "spec", "template", "spec", "initContainers")
+
+		if kind == "CronJob" {
+			containers, _, _ = unstructured.NestedSlice(u.Object, "spec", "jobTemplate", "spec", "template", "spec", "initContainers")
+		}
+	} else {
+		containers, _, _ = unstructured.NestedSlice(u.Object, "spec", "template", "spec", "containers")
+
+		if kind == "CronJob" {
+			containers, _, _ = unstructured.NestedSlice(u.Object, "spec", "jobTemplate", "spec", "template", "spec", "containers")
+		}
```

Comment:
Since container names are unique, maybe instead of passing down `isInitContainer` you could just append both initContainers and containers together and search for the name?